### PR TITLE
lk/pluginify translation

### PR DESCRIPTION
**Pull request by leangseu-edx**
- [x] move feedback widget and translation selection to plugin folder
- [x] take advantage of using https://github.com/openedx/frontend-plugin-framework/
- [x] clean up config linting/test
- [x] write a more verbose of readme

### Future plan
- replace `@plugin-framework` to the package from https://github.com/openedx/frontend-plugin-framework/
- contribute the `usePluginsCallback` and `registerOverrideMethod` to https://github.com/openedx/frontend-plugin-framework/

### CODE REVIEW

File: plugins/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- The way for how I deal with override method is still wonky
+  - The redux still require middleware to ignore the plugin's action from serializing
+  - I am not sure how it behave with useCallback, useMemo, ...etc
+  - There are still open question on how to write it properly
+- `@plugin-framework` will be replace by https://github.com/openedx/frontend-plugin-framework/. I will leave it there for now.
+- I am still unsure about writing tests for this
+
+## Current work that should consider core part and extendable for the future plugin framework
+
+- `usePluingsCallback` is the callback supose to be some level of equality to be using `React.useCallback`. It would try to execute the function, then any plugin that try `registerOverrideMethod`. The order of the it being run isn't the determined. There are a couple things I want to add:
```
Comment (BrandonHBodine) [inline plugins/README.md:16]: typo? Not sure if it matters if this is all temporary.

File: plugins/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## How to develop plugin
```
Comment (BrandonHBodine) [inline plugins/README.md:1]: Nice Readme; thanks for including this.

File: src/courseware/course/sequence/Unit/hooks/useIFrameBehavior.js [hunk 3]
```diff
@@ -5,9 +5,9 @@ import { useDispatch } from 'react-redux';
 import { StrictDict, useKeyedState } from '@edx/react-unit-test-utils';
 import { logError } from '@edx/frontend-platform/logging';
 
-import { fetchCourse } from '../../../../data';
-import { processEvent } from '../../../../../course-home/data/thunks';
-import { useEventListener } from '../../../../../generic/hooks';
+import { fetchCourse } from '@src/courseware/data';
```
Comment (BrandonHBodine) [inline src/courseware/course/sequence/Unit/hooks/useIFrameBehavior.js:8]: What library introduces this syntax`@src/...`?

File: src/courseware/course/sequence/Unit/index.jsx [hunk 4]
```diff
@@ -24,34 +24,28 @@ const Unit = ({
...
-  const { language, wholeCourseTranslationEnabled } = useModel('courseHomeMeta', courseId);
   const isProcessing = unit.bookmarkedUpdateState === 'loading';
   const view = authenticatedUser ? views.student : views.public;
-  const { selectedLanguage, setSelectedLanguage } = useSelectLanguage({
-    courseId,
-    language,
-  });
 
-  const iframeUrl = useMemo(() => getIFrameUrl({
+  const getUrl = usePluginsCallback('getIFrameUrl', () => getIFrameUrl({
```
Comment (BrandonHBodine) [inline src/courseware/course/sequence/Unit/index.jsx:30]: Nice! This is a seemingly elegant solution. I wonder what kinds of perf impacts this might have?

**Review by BrandonHBodine — APPROVED** [review]
This is looking great! Left some general questions but nothing of significance.

**Review by Rodra — APPROVED** [review]

File: src/courseware/course/sequence/Unit/hooks/useIFrameBehavior.js (hunk 3 above)
Comment (leangseu-edx) [reply src/courseware/course/sequence/Unit/hooks/useIFrameBehavior.js:8]: `@src`, `@plugins` and `@plugin-framework` are aliases. It make the code easier to move compare to relative path.

**Pull request merged**